RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,17 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    edm::LogError("PhotonProducer")
+      << "Error! Can't get the preshowerEcalHits";
+    validPreshowerRecHits=false;
+  }
```

Comment:
The job should just fail if a configured input is not present.
If the goal is to allow running without preshowerRecHits, it should be configured accordingly:
- add a config flag usePreshowerRecHits
- treat an empty input tag as serving this choice